### PR TITLE
examples: fixed fault-injection delay demo

### PR DESCRIPTION
**Pull request by cpakulski**
Description:
fault injection example did not work for delay option. After gdb debugging it was discovered that fault injection config requires `delay` part to be present. Without `delay` part Envoy assumes that Delay is disabled and does not scan RTDS for delay config updates.

Risk Level:
Low: 

Testing:
Did manual testing as described in https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/fault_injection

Docs Changes:
No

Release Notes:
No

Fixes: #11095

### CODE REVIEW

File: examples/fault-injection/docker-compose.yaml [hunk 1]
```diff
@@ -13,6 +13,8 @@ services:
     ports:
       - 9211:9211
       - 9901:9901
+    environment:
+      ENVOY_UID: 0
```
Comment (snowp) [inline examples/fault-injection/docker-compose.yaml:17]: why is this necessary?

Comment (dio) [reply examples/fault-injection/docker-compose.yaml:17]: FWIW https://github.com/envoyproxy/envoy/pull/11523

File: examples/fault-injection/envoy.yaml [hunk 2]
```diff
@@ -36,6 +36,11 @@ static_resources:
                 percentage:
                   numerator: 0
                   denominator: HUNDRED
+              delay:
```
Comment (alyssawilk) [inline examples/fault-injection/envoy.yaml:39]: This is a great find, thanks for tracking it down!

I do wonder if we should tweak the fault filter to delay / abort if there's a runtime override even if the proto doesn't include the config.  Even though it's pretty clearly documented, I can imagine it confusing folks.
cc @mattklein123 

either way it shouldn't block this PR, more musing if we should file a tracking issue.

File: examples/fault-injection/docker-compose.yaml (hunk 1 above)
Comment (cpakulski) [reply examples/fault-injection/docker-compose.yaml:17]: It was necessary to give Envoy permission to access `/dev/stdout`. Without that setting Envoy container exists complaining that it cannot write logs to `/dev/stdout` (specified in envoy.yaml as sink for logs).

Comment (alyssawilk) [reply examples/fault-injection/docker-compose.yaml:17]: Optional, bonus points for commenting that - I ran into that trying to repro and it'd be lovely if a local grep for stdout turned up the fix :-)

File: examples/fault-injection/envoy.yaml (hunk 2 above)
Comment (mattklein123) [reply examples/fault-injection/envoy.yaml:39]: I don't recall the history of why this is implemented the way it is. I agree it's potentially confusing. I think the idea is we want the user to opt-in explicitly to the config they want but I don't feel super strongly about it. @Augustyniak any thoughts here?

File: examples/fault-injection/docker-compose.yaml (hunk 1 above)
Comment (cpakulski) [reply examples/fault-injection/docker-compose.yaml:18]: Added comment to explain why Envoy must run as root.

File: examples/fault-injection/envoy.yaml (hunk 2 above)
Comment (Augustyniak) [reply examples/fault-injection/envoy.yaml:39]: I think explicitness here is good since we support injecting faults via different methods including [HTTP headers](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/fault_filter#controlling-fault-injection-via-http-headers) and we want these faults to be injected only if our Envoy is explicitly configured to accept them.

- Without Envoy requiring people to opt-in with this config, it may be confusing as to what kind of faults a given Envoy may potentially inject - the ones controlled via runtime or via HTTP headers. With config in, it becomes clear what’s supported.
- If we decide that we don’t want to require a config for faults controlled via runtime we should probably do the same for faults controlled with HTTP headers which is risky because of security implications.

For all of the aforementioned reasons, I would leave the presence of the config as the required step of `FaultFilter`’s setup.

For reference, an example of a config that supports delay faults via HTTP headers:
```
delay:
  header_delay: {}
  percentage:
  numerator: 100
```

**Comment by cpakulski** [comment]
@alyssawilk I think this is ready for merge.
